Added updated DartboardView and UI and removed painter_example

ethanshattuck opened the pull request:
Added updated DartboardView and UI files and removed painter_example

ethanshattuck committed:
Added updated DartboardView and UI and removed painter_example

Jedimaster4559 commented on dartboard/DartboardView.py:
```diff
@@ -0,0 +1,299 @@
+from PySide2.QtWidgets import QApplication, QWidget
```
Can you add Doxygen documentation to each of your files/classes/methods? Information on how to do this can be found [here](https://www.doxygen.nl/manual/docblocks.html#pythonblocks). I'm working on setting up automatic documentation publishing as well, but for now, you can see the results of your doxygen documentation by downloading the `Dartboard Documentation` artifact from the main workflow.

Jedimaster4559 replied:
Don't worry too much about the formatting of the doxygen docs for now. I'm still working on figuring out how to format everything so the built documentation is pretty and easy to read. I'll make a pass over everything to update the formatting of the comments to work well with doxygen once I have all that figured out. For now, just having that documentation will be super helpful (even if it's a little tougher to navigate).

ethanshattuck replied:
Ok will do.

Jedimaster4559 commented on dartboard/DartboardView.py:
```diff
@@ -0,0 +1,299 @@
+from PySide2.QtWidgets import QApplication, QWidget
+from PySide2.QtWidgets import *
```
Do you actually need to import * here? I notice between this and the line above, you're importing from `QtWidgets` twice?

ethanshattuck replied:
You can either specifically import all of the widgets you need, or import all using *, I can fix that

Jedimaster4559 commented on dartboard/DartboardView.py:
```diff
@@ -0,0 +1,299 @@
+from PySide2.QtWidgets import QApplication, QWidget
```
Should we create a `views` directory or something like that to help with general organization? I think if we put too much in the top level directory things could start getting pretty cluttered.

ethanshattuck replied:
I don't think that's necessary. That is the most complicated control we will have. Everything else will be part of the UI file I think.

Jedimaster4559 replied:
I think for now then we can leave it, but eventually things might need to change home. If we have more than 3-4 graphics files at this level, we should move some into their own package.

Jedimaster4559 commented on dartboard/DartboardView.py:
```diff
@@ -0,0 +1,299 @@
+         print(event.pos())
+
+         #
+         hit = DartboardHit(x=event.pos().x(), y=event.pos().y())
+         hit.save()
+         print(DartboardHit.objects.all())
```
Can we remove these calls to the backend for now? Now that we are no longer using the demo code for this, I think it's best we leave this direct database manipulation out.

ethanshattuck replied:
Sure.

Jedimaster4559 commented on dartboard/DartboardView.py:
```diff
@@ -0,0 +1,299 @@
+         self.scene.addEllipse(self.WindowPosX + event.pos().x(), self.WindowPosY + event.pos().y(), 3,3, QPen(), QBrush(Qt.white))
+
+    #this function is called when the window is resized
+    def resizeEvent(self, event):
```
So, another question about this class in general. When the window gets re-sized, does that change the x and y values of the previous dartboard hits? Could this potentially cause some issues?

For Example: If I store all my dartboard hits to the database off of their x and y coordinates, then resize the window dramatically, will those x and y coordinates still represent the correct spot on the  dartboard?

ethanshattuck replied:
So I set this up to scale proportionately. You would record dartboard hits by their offsets from the center point and the radius of the screen. Then, when they are added later with a different sized screen, the offsets will be scaled based on the difference in radius to place the points accurately.

Jedimaster4559 commented on dartboard/DartboardView.py:
```diff
@@ -0,0 +1,299 @@
+        #recall the drawing function to replace the circles and points with the new size
+        self.DrawRegions()
+
+        #tbh idek what this does
```
Explanation:<br>
This class inherits off of some super class within the QT library (`QGraphicsView` I think). doing `super().resizeEvent(event)` performs a call of the resize event on the super class. This is important to do because by creating your own method called `resizeEvent` you override the default behaviour of the resize Event that is specified within `QGraphicsView`. That default behaviour still includes a bunch of important stuff, so it's necessary that we tell our version to still do all of the default behaviour on top of the extra steps we have added.

Let me know if you have any questions about this, if not, please update the comment.

ethanshattuck replied:
I'm aware of what the super class does.. I just didn't see any effect on the code when this call was commented out or not.

Jedimaster4559 replied:
Oh, I see, I think we leave it then. It could be doing something important on the backend that isn't readily visible to us right now. I think we still should remove the comment.

Jedimaster4559 commented on dartboard/DartboardView.py:
```diff
@@ -0,0 +1,299 @@
+        self.setScene(self.scene)
+
+        #set variables and constants
+        self.WindowSizeX = 300
```
Can you add some super basic documentation on the really important variables here? (specifically, clarifying region vertices would be nice).

Jedimaster4559 commented on dartboard/DartboardView.py:
```diff
@@ -0,0 +1,299 @@
+        color = True
+        for i in self.inner_regions:
+            if (color):
+                self.scene.addPolygon(i, QPen(), QBrush(Qt.white))
```
Not super relevant here, but eventually it might be fun to make these colors configurable.

Jedimaster4559 commented on dartboard/DartboardView.py:
```diff
@@ -0,0 +1,299 @@
+
+            
+
+        self.triples_regions = []
```
Can you add a little more documentation to the second half of this method (more along the lines of what you have for the rest of this class)

Jedimaster4559 commented on dartboard/DartboardView.py:
```diff
@@ -0,0 +1,299 @@
+#create application and window
+#app = QApplication(sys.argv)
+#window = Window()
+#sys.exit(app.exec_())
```
I can't remember if I was the one to delete this code or not, but if we don't need it anymore, can we delete it?

Jedimaster4559 commented on dartboard/TestUI.ui:
```diff
@@ -0,0 +1,44 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
So, it took me a few minutes to figure this out. If we can avoid names including the work "test" for files that would be preferable. Only files that actually perform actually unit, functional, ui, or other testing should match that naming scheme. Additionally, those should be somewhere inside the testing directory.

Jedimaster4559 commented on dartboard/TestUI.py:
```diff
@@ -0,0 +1,49 @@
+# -*- coding: utf-8 -*-
+
+################################################################################
```
I realize this is an autogenerated file, but if we do the views thing mentioned above, it would be really nice if we could move this to that views directory.

Jedimaster4559 commented on dartboard/DartboardView.py:
```diff
@@ -0,0 +1,299 @@
+        pi = math.pi
+        points = [QPoint(math.cos(2*pi/n*x)*r + self.WindowPosX + (self.WindowSizeX / 2.0), math.sin(2*pi/n*x)*r + self.WindowPosY + (self.WindowSizeY / 2.0)) for x in range(0,n+1)]
+        for p in points:
+            self.rotate(p, -pi/19)
+        return points
+
+    def rotate(self, point, angle):
+        centerX = self.WindowPosX + (self.WindowSizeX / 2.0)
+        centerY = self.WindowPosY + (self.WindowSizeY / 2.0)
+        point.setX(centerX + math.cos(angle) * (point.x() - centerX) - math.sin(angle) * (point.y() - centerY))
+        point.setY(centerY + math.sin(angle) * (point.x() - centerX) + math.cos(angle) * (point.y() - centerY))
```
Please update to something like this (performs an actual rotation of the points) (also note the update to `-pi/20`)

```suggestion
            self.rotate(p, -pi/20)
        return points

    def rotate(self, point, angle):
        # Get Center of Dartboard
        centerX = self.WindowPosX + (self.WindowSizeX / 2.0)
        centerY = self.WindowPosY + (self.WindowSizeY / 2.0)
        
        # Calculate important trig information
        sin = math.sin(angle)
        cos = math.cos(angle)
        
        # Offset points so the center is the origin
        px = point.x() - centerX
        py = point.y() - centerY
        
        # Calculate new rotated points
        xnew = (px * cos) - (py * sin)
        ynew = (px * sin) + (py * cos)
        
        # Update actual point data
        point.setX(xnew + centerX)
        point.setY(ynew + centerY)
```

Jedimaster4559 reviewed (changes requested):
# Summary
I've attached a number of comments above. Please handle those when you have a chance and then re-request a review from me. Really great work getting all of this built so quickly!

# Text/Fonts
So, I know you mentioned this a couple times, and it isn't really in the scope of this update but I figured it would be nice to have documented here. I think the numbers still feel a little clunky. I'm not sure exactly how you have the positions and the sizing of the buttons setup but it would be nice if they scaled and were positioned a little better at small vs. large sizes. I also think it might be nice to use a different font that has more weight to it (not entirely sure about this though. Something about the font doesn't feel right though). It could also be that a bolded sans-serif font would look a lot stronger.

# Testing
I ran the application from my dev environment and everything seemed to work fine from there. I also ran pyinstaller locally and that was able to properly build the executable. I didn't check the pipeline version since that shouldn't work until we take care of #2.

ethanshattuck committed:
Modified class and ui names to make more sense

Removed test from names of classes and ui files. Created a "Hub.py" class that will be the main area for operating between UIs.

ethanshattuck committed:
Added initial View Manager called Hub.py

ethanshattuck committed:
Added navigation to configuration before game

ethanshattuck committed:
Added print statements for all data in StartGameConfiguration window

ethanshattuck committed:
Added anti-aliasing to Dartboard

ethanshattuck committed:
Added layout for Scorer based on Alex's Mockup

ethanshattuck committed:
Added dark theme and realignment of the Scoreboard and Scorer GUIs

21sean reviewed (approved):
Looks great!

Jedimaster4559 commented on dartboard/dartboard_main.py:
```diff
@@ -4,14 +4,43 @@
+
+
+if __name__ == "__main__":
+    app = QApplication(sys.argv)
```
Frontend Code in main class. Can we move  this out of this?

Jedimaster4559 reviewed (approved):
Move GUI code  into directory and address my other comment attached.

ethanshattuck committed:
Added views folder and moved UI code out of main
